File: leetcode/235.py
from utils import *

class Solution:
  def lowestCommonAncestor(self, root:TreeNode, p:TreeNode, q:TreeNode) -> TreeNode:
    stack = [root]
    while stack:
      node = stack.pop()
      if (p.val < node.val and q.val < node.val): stack.append(node.left)
      elif (p.val > node.val and q.val > node.val): stack.append(node.right)
      else: return node

File: leetcode/235.js
const lowestCommonAncestor = function(root, p, q) {
  let stack = [root];
  while(stack.length) {
    let node = stack.pop();
    if(node == null) continue;
    if (
      (p.val <= node.val && node.val <= q.val) ||
      (q.val <= node.val && node.val <= p.val)
    ) return node;
    stack.push(node.right)
    stack.push(node.left)
  }
  return root;
};
